### ISSUE TITLE
Memory leak detection with leak_tracker_flutter_testing

### ISSUE THREAD

**Issue by kosratdev** [imaNNeo/fl_chart#1975]
**Describe the bug**
After adding the `leak_tracker_flutter_testing` dependency to the dev dependencies to detect memory leaks while running the tests. I found two memory leaks that happens multiple times in different tests. They are not disposed objects as listed below: 
- **PanGestureRecognizer** object in `render_base_chart.dart` file
- **TapGestureRecognizer** object in `render_base_chart.dart` file
- **LongPressGestureRecognizer** object in `render_base_chart.dart` file

The above memory leaks can be fixed easily just by disposing them in the `detach` method like below: 
```dart
  @override
  void detach() {
    _validForMouseTracker = false;
    panGestureRecognizer.dispose();
    tapGestureRecognizer.dispose();
    longPressGestureRecognizer.dispose();
    super.detach();
  }
```
Furthermore, it has some other memory leaks, which are hard for me to fix as I'm not very familiar with the source code architecture.
- **TransformationController** in `axis_chart_scaffold_widget.dart` file.
- **TransformationController** in `custom_interactive_viewer.dart` file.

**To Reproduce**
- Add `leak_tracker_flutter_testing` to the dev dependancies 
```
dev_dependencies:
  ...
  leak_tracker_flutter_testing: any
```
- Create `flutter_test_config.dart` file in the root of the test folder
```
import 'dart:async';

import 'package:leak_tracker_flutter_testing/leak_tracker_flutter_testing.dart';

Future<void> testExecutable(FutureOr<void> Function() testMain) async {
  LeakTesting.enable();
  LeakTesting.settings = LeakTesting.settings
    .withIgnored(createdByTestHelpers: true);

  await testMain();
}
```
- run tests to see detected memory leaks. 

**Screenshots**
If applicable, add screenshots, or videoshots to help explain your problem.

**Versions**
 - Flutter Version: 3.35.1
 - FlChart Version: 1.0.0

I want to create a pull request to fix those memory leaks, but I can't fix the `TransformationController` leaks unless you help me.

**Comment by imaNNeo** [comment]
Hi @kosratdev 
Good catch and thanks for your detailed issue

At the moment, I don't have time to test them.
Can you please check if those `.dispose()` calls fix the issue and if they don't break anything, open a PR?

For the other one (transformer controller), we need more investigation.
Maybe @Peetee06 can help us on that

**Comment by peter-trost** [comment]
Hey there, thanks for flagging these memory leaks!

I don't have any capacity to look into this myself right now, but am happy to answer questions if there are specific ones..

**Comment by harikrishna-au** [comment]
As suggested by @imaNNeo and @Peetee06, I will dispose the gesture recognizers in render_base_chart.dart within the detach() method, and ensure proper disposal of TransformationController in the relevant State.dispose() methods. I will verify the fixes using leak_tracker_flutter_testing and then open a PR referencing this issue.
